### PR TITLE
[IMP] mass_mailing: Basic theme(Plain text) template should look proper in gmail

### PR DESCRIPTION
**Pull request by hiralvbhavsar**
Description of the issue/feature this PR addresses:

Task: https://www.odoo.com/web?#id=38858&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.d2dd5a8147f7ab4da9bf60b89a5479b4

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jat-odoo** [comment]
LGTM

File: addons/mass_mailing/static/src/js/mass_mailing_editor.js [hunk 1]
```diff
@@ -143,6 +143,31 @@ $.summernote.eventHandler.modules.popover.update = function ($popover, oStyle, i
...
+     *
+     * @override
+     */
+    cleanForSave: function () {
+        this._super.apply(this, arguments);
+        var $basicTheme = this.$editable.find('.o_layout.o_basic_theme');
+        if ($basicTheme.length) {
+            $basicTheme.css('padding', '0px');
+        }
+    },
```
Comment (qsm-odoo) [inline addons/mass_mailing/static/src/js/mass_mailing_editor.js]: Why is this needed in JS ?

File: addons/mass_mailing/views/snippets_themes.xml [hunk 2]
```diff
@@ -12,7 +12,11 @@
                      data-nowrap="1"
                      data-img="/mass_mailing/static/src/img/theme_imgs/basic_thumb"
                      data-images-info='{"logo": {"format": "png"}}'>
-                    <t t-call="mass_mailing.theme_basic_template"/>
+                    <div>
+                        <p>
+                            <br/>
+                        </p>
+                    </div>
```
Comment (qsm-odoo) [inline addons/mass_mailing/views/snippets_themes.xml]: I think we can keep the basic template view and put that simple DOM in it

File: addons/mass_mailing/static/src/js/mass_mailing_editor.js (hunk 1 above)
Comment (hbh-odoo) [reply addons/mass_mailing/static/src/js/mass_mailing_editor.js]: As per the spec we also have to maintain the padding of text in gmail. Please look at [https://drive.google.com/file/d/0B1uIL9E_zXrrb3dCZVpZcEIwU3c/view](url)
If we remove padding from template then not getting proper preview of template, i mean all texts are placing at left corner like [https://drive.google.com/file/d/1o4mZ4IRlzmwi4ZJybPLPtoYlVg4McCNX/view?usp=drivesdk](url)
That's why we've removed padding on save and reset it in edit mode to maintain preview of template as well as in gmail.

File: addons/mass_mailing/views/snippets_themes.xml (hunk 2 above)
Comment (hbh-odoo) [reply addons/mass_mailing/views/snippets_themes.xml]: Now we are removing the default text then cann't we move that blank part in this instead of calling that template?

File: addons/mass_mailing/static/src/js/mass_mailing_editor.js (hunk 1 above)
Comment (qsm-odoo) [reply addons/mass_mailing/static/src/js/mass_mailing_editor.js]: I think that what we need to do is:
- Removing the padding of the `.o_layout` element (content element)
- Adding padding to the `#editable_area` element  (editor UI element)

Both in LESS (`theme_default.less` and `mass_mailing.ui.less` I suppose). Should not it solve the problem without any JS ?

File: addons/mass_mailing/views/snippets_themes.xml (hunk 2 above)
Comment (qsm-odoo) [reply addons/mass_mailing/views/snippets_themes.xml]: I think I prefer consistency. As we are calling a template for the `default` theme and the enterprise themes, we should call one here too, even if it is nearly empty. This also allows users to modify that default template if they want to.

File: addons/mass_mailing/static/src/js/mass_mailing_editor.js (hunk 1 above)
Comment (hbh-odoo) [reply addons/mass_mailing/static/src/js/mass_mailing_editor.js]: It solve the problem but not proper preview of template. see [https://drive.google.com/file/d/1JoGKPP8c_omT89QA6Vqd6FaACWX2LzIv/view?usp=drivesdk](https://drive.google.com/file/d/1JoGKPP8c_omT89QA6Vqd6FaACWX2LzIv/view?usp=drivesdk)
is it ok?

Comment (qsm-odoo) [reply addons/mass_mailing/static/src/js/mass_mailing_editor.js]: Well can't we make the `#editable_area` element white as well (instead of explicitly transparent I think)?

Comment (hbh-odoo) [reply addons/mass_mailing/static/src/js/mass_mailing_editor.js]: yes...we can

File: addons/mass_mailing/static/src/less/themes/theme_default.less [hunk 3]
```diff
@@ -53,7 +54,6 @@
     width: 94%;
```
Comment (qsm-odoo) [inline addons/mass_mailing/static/src/less/themes/theme_default.less]: This should be `width: 100%;` now

File: addons/mass_mailing/static/src/less/themes/theme_default.less [hunk 4]
```diff
@@ -27,7 +27,8 @@
 // ===== Editor =====
 #web_editor_inside_iframe {
     #wrapwrap > main #editable_area {
-        background-color: transparent;
+        background-color: white;
+        padding: 24px 3%;
```
Comment (qsm-odoo) [inline addons/mass_mailing/static/src/less/themes/theme_default.less]: I think we can remove the `3%` and use `24px` all around.

File: addons/mass_mailing/static/src/less/themes/theme_default.less [hunk 5]
```diff
@@ -51,10 +52,9 @@
 .o_layout {
     overflow: hidden;
     box-sizing: content-box;
-    width: 94%;
+    width: 100%;
     min-height: 100%;
     background-color: @o_mm_def_color_alpha;
-    padding: 24px 3%;
```
Comment (qsm-odoo) [inline addons/mass_mailing/static/src/less/themes/theme_default.less]: In fact, the top and bottom paddings are needed for the design of the other themes, but I will change that myself and merge :)

**Comment by qsm-odoo** [comment]
Merged https://github.com/odoo/odoo/commit/66aefb1723c336cc1c61fb32926ed926ba9e29e4
